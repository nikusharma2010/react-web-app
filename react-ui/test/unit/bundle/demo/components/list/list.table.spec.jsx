import React from "react";
import { create } from "react-test-renderer";
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter } from 'react-router-dom';
import ListTable from "../../../../../../src/bundle/demo/components/list/list.table";
Enzyme.configure({ adapter: new Adapter() });

describe("List Component", () => {
    test("it matches the snapshot", () => {
        const props = {
            usersList: [
                {
                    "id": "7",
                    "firstName": "Niku",
                    "lastName": "Sharma",
                    "email": "7@thesimpsons.com",
                    "mobile": "007"
                }]
        };
        const component = create(<BrowserRouter><ListTable {...props} /></BrowserRouter>);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
