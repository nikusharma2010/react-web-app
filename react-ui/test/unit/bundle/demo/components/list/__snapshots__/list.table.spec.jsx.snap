// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component it matches the snapshot 1`] = `
<div
  className="styles__Row-sc-18xwbcb-2 eCIkQt"
  id="user_id_7"
>
  <div
    className="styles__Col-sc-18xwbcb-1 FOjru"
  >
    <div
      className="styles__NameLabel-sc-18xwbcb-4 eFarEQ"
    >
      N
      S
    </div>
  </div>
  <div
    className="styles__Col-sc-18xwbcb-1 eqLmFH"
  >
    <div>
      <b>
        Niku Sharma
      </b>
    </div>
    <div>
      007
    </div>
    <div>
      7@thesimpsons.com
    </div>
  </div>
  <div
    className="styles__Col-sc-18xwbcb-1 bLiUMj"
  >
    <a
      href="/view/7"
      onClick={[Function]}
    >
      <div
        className="styles__Button-sc-18xwbcb-5 styles__ButtonPrimary-sc-18xwbcb-6 YFSZa"
      >
        View
      </div>
       
    </a>
  </div>
</div>
`;
